### ISSUE TITLE
Curl error

### ISSUE THREAD

**Issue by sajjadrad** [skovmand/mailchimp-laravel#16]
Hi,
I have this error while posting a request to mailchimp:
ErrorException in Mailchimp.php line 198: curl_setopt() expects parameter 1 to
 be resource, integer given

I checked out mailchimp master class construct, ch initiated successfully but in call method ch is 0!
I'm using this package in a job. and initiated mailchimp class with app("Mailchimp").

**Comment by skovmand** [comment]
I'm don't know what that problem is, but I don't think it lies within this service. This package really only gives you an instance of the Mailchimp client, not much else, so I think the problem lies in the way you use Mailchimp or Laravel Jobs. 

But let me know if you find an error in this service.

**Comment by dewwwald** [comment]
@sajjadrad  I also had this issue. I am not sure how to fix it. 

- The issue appears to be job specific. (at least as far as this feed is concerned)
- The IOC container call method seem to lose the data of the curl instance. (e.g. $this->ch is 0)

My thinking that is unconfirmed at this point. The IOC container call method is called at a time where the curl instance is now closed (probably for security reasons).
http://stackoverflow.com/questions/6887104/why-am-i-getting-http-code-0-from-curl#answer-6969738

Suggested fix:
- Capture data in the constructor
- init curl instance when it is called

 I actually did this on you package and it works for me. I'll make a PR in a bit.
